### DISCUSSION TITLE
Changing frontmatter based on logic within a post

### DISCUSSION THREAD

**Discussion by splotsh** [jekyll/jekyll#8547] [Q&A]
Hi,

I have a large number of pages (1000+), and I'm trying to prevent some of these pages from being indexed by Google as they are mostly empty. Using jekyll-sitemap I was hoping there would be a way I could change/use a variable in frontmatter of all my pages that would set "sitemap:" to true or false depending on some logic I have in my page content.

I've looked at jekyll-liquify to see if it could solve my problem by allowing me to use variables in frontmatter, but the way this plugin does that is not sufficient for me. I'm happy to customize jekyll-sitemap if someone can think of a solution here.

Thanks!

**Comment by fauno** [comment]
Apparently jekyll-sitemap already does what you want :) 

https://github.com/jekyll/jekyll-sitemap#exclusions

Edit: you could also generate a robots.txt file that disallows indexing this page with a "query" like this `{% assign disallowed_posts = site.post | where: 'sitemap', false %}`, or add `noindex` to a conditional meta tag in the `<head>`.

**Reply by splotsh** [reply]
Not quite :) Unfortunately I can't manually set `sitemap: false` on all the pages I want to prevent getting indexed. I need to set it false or true depending on the existence of a variable in the post to make this option work for me with the amount of pages I've got.
A conditional meta tag in the `<head>` I tried as well, however since my variable is declared in an include my `{% if sitemap_variable %}`sitting in `<head>` gets called before the variable gets declared.

Anything else I could do ?

**Comment by fauno** [comment]
But `sitemap: false` is a variable in each post, if you mean you already
have a variable and it's not called `sitemap`, you can also write a hook
to modify each post's data.

**Reply by bheinkel** [reply]
It's a frontmatter variable, not a variable I can manipulate from within the post using liquid - which is what I really need.
Anyway, I've solved this now by injecting a noindex meta tag into the body of the post when a certain condition is met. Google apparently respects this tag even if it isn't in the header. Thanks for your help

**Comment by ashmaroli** [comment]
@splotsh If you have a `sitemap.xml` at the root of your site's source directory, this plugin will not generate the default sitemap.
So, you're free to use the sitemap Liquid template in this repository (or the plugin installed locally) as a starting point and customize it as desired.
